### PR TITLE
Add events and remove incidents from Concepts index page

### PR DESCRIPTION
**Pull request by discdiver**
Currently, Events is in the Concepts nav bar section and Incidents is in the Cloud navbar section.
On the Concepts index page Incidents was referenced and Events was not.

This PR rectifies the problem by removing Incidents from the index page and adding Events.

Also cleans up a few redirect rules.

### Checklist

- [ ] This pull request references any related issue by including "closes `<link to issue>`"
	- If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect/issues/new/choose) first.
- [X] This pull request includes tests or only affects documentation.
- [X] This pull request includes a label categorizing the change e.g. `maintenance`, `fix`, `feature`, `enhancement`, `docs`.
  

For documentation changes:

- [ ] This pull request includes redirect settings in `netlify.toml` for files that are removed or renamed.

For new functions or classes in the Python SDK:

- [ ] This pull request includes helpful docstrings.
- [ ] If a new Python file was added, this pull request contains a stub page in the Python SDK docs and an entry in `mkdocs.yml` navigation.

### CODE REVIEW

File: docs/concepts/index.md
Comment (serinamarie) [inline docs/concepts/index.md:27]: I think I'm missing context on what's happening :)

Comment (discdiver) [reply docs/concepts/index.md:27]: Sorry, I should have put in the description.

Currently, Events is in the Concepts nav bar section and Incidents is in the Cloud navbar section.
On the Concepts index page Incidents was referenced and Events was not. 

This PR rectifies the problem by removing Incidents from the index page and adding Events.

Comment (serinamarie) [reply docs/concepts/index.md:27]: oh I see ! but shouldn't it actually be the case that neither should be there and both should be in the cloud toc and nav bar?

Comment (discdiver) [reply docs/concepts/index.md:27]: I think it's a gray area for events. @billpalombi laid out some nice criteria for what fits in the Concepts section of the docs:

- If you were explaining how Prefect works, would you need to use this word?
- If there are a diagram of Prefect components, would this word be on it?
- Does this word have unique meaning in Prefect relative to other systems?

I think the answers to all three are it depends/maybe/somewhat. :) 

I could lean toward moving Events  to the Cloud section.

Comment (serinamarie) [reply docs/concepts/index.md:27]: i'd argue if it's a cloud-only feature, then it shouldn't be necessary to explain how prefect works, both of which applies to events and incidents

Comment (discdiver) [reply docs/concepts/index.md:27]: With Automations in Concepts, maybe best to keep Events there too, as they should probably stick together. I could merge this PR and then open another if we decided to move other items to Cloud.

Comment (serinamarie) [reply docs/concepts/index.md:27]: Incidents is the only cloud feature that ended up in the Prefect concepts, so my vote would be to just remove Incidents from there. Happy to defer to others on that.

Comment (discdiver) [reply docs/concepts/index.md:27]: @serinamarie do I understand correctly that you are saying that this PR is good as it stands?

Comment (kevingrismore) [reply docs/concepts/index.md:27]: I think @serinamarie is saying that incidents being under the Cloud section only makes sense if other cloud-exclusive features are there too. I do think it's a little confusing right now because it's hard to differentiate what's a Prefect "concept" and what's a feature of Cloud, despite the badges. It doesn't feel quite right to me that the Concepts section keeps growing every time we add a cloud-exclusive capability.

Comment (serinamarie) [reply docs/concepts/index.md:27]: yeah, I'm saying this PR could probably just remove incidents from the concepts page and add events to the cloud page

Comment (billpalombi) [reply docs/concepts/index.md:27]: I agree with @serinamarie

Comment (discdiver) [reply docs/concepts/index.md]: Thank you all. I've moved events to cloud.

**Review by serinamarie — APPROVED** [review]
LGTM! 
Screenshot 2024-01-24 at 11 57 33 AM
Screenshot 2024-01-24 at 11 57 51 AM

**Pull request merged**
